frontend/src/main/scala/bloop/engine/caches/ResultsCache.scala

```diff
@@ -177,7 +177,7 @@ object ResultsCache {
       if (!cleanOrphanedInternalDirs) Task.unit
       else {
         val internalClassesDir =
-          CompileOutPaths.createInternalClassesRootDir(project.genericClassesDir)
+          CompileOutPaths.createInternalClassesRootDir(project.out)
```

Comment:
Could you provide concrete values for `project.out` and `project.genericClassesDir`? Is one of them `bloop-internal-classes`?

Reply:
`project.out` is the currently used one for example `/.bloop/input3/` while `genericClassesDir` is `.bloop/input3/scala-3/classes`, which is actually defined in the json file.

Reply:
I'm actually a bit surprised by this, I thought it _would_ use the one specific to the scala version and therefore the nested on in `.bloop/input3/scala-3/classes`.

Reply:
That is kind of redundant since you can't have multiple Scala versions for one project, so I don't think it's really useful. Also, it's a smaller change to just ignore that directory :S

Reply:
also, it seems that we always use `bloop-bsp-clients-classes` always instead of `classesDir`, so this looks legit as if we had only a single output we could have issues with mutiple clients. (whether we could have mutiple BSP client for the same workspace is another thing altogether.)

Reply:
Can you add scaladoc which will explain what orphaned directories are? Giving some concrete examples you did int his thread will awesome

Reply:
Ok, more digging it seems the is an option for the tool to use managed classes directory, but there is even more logic on top of it to dermine it.

Reply:
> That is kind of redundant since you can't have multiple Scala versions for one project, so I don't think it's really useful. Also, it's a smaller change to just ignore that directory :S

ah ok cool, thanks for the explanations 😄

Reply:
I added scaladoc to this method. I will probably dig in a bit more later on.